### ISSUE TITLE
socket.io client is missing removeAllListeners() in typescript definitions

### ISSUE THREAD

**Issue by mrtnlrsn** [socketio/socket.io#4174]
**Describe the bug**
socket.io client is missing removeAllListeners() in typescript definitions.

**To Reproduce**

Socket.IO client version: `4.1.2`

*Client*

```ts
import { io } from 'socket.io-client'
const socket = io('http://localhost:5090')
socket.on('hi', () => console.log('hi'))
socket.removeAllListeners('hi')
```

compile error:
```
mla.ts:4:8 - error TS2339: Property 'removeAllListeners' does not exist on type 'Socket<DefaultEventsMap, DefaultEventsMap>'.

4 socket.removeAllListeners('hi')
```

java script code works fine:
```js
const io = require('socket.io-client').io
const socket = io('http://localhost:5090')
socket.on('hi', () => console.log('hi'))
socket.removeAllListeners('hi')

```

**Expected behavior**
Typescript code should compilie

**Platform:**
 - Device: [Intel PC]
 - OS: [Ubuntu 20]
 - node 12.18, all packages uptodate

**Additional context**

**Comment by darrachequesne** [comment]
Hi! It should be fixed in `@types/component-emitter@1.2.11`, you should be able to update the definitions in your project with `npm update --depth 9999 @types/component-emitter`, as `socket.io-client@4.1.2` includes version [`^1.2.10`](https://github.com/socketio/socket.io-client/blob/4.1.2/package.json#L30).

Related: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/55968

**Comment by mrtnlrsn** [comment]
@darrachequesne You are correct.
Sorry, I thought I was on the latest version.
